### ISSUE TITLE
headless problem

### ISSUE THREAD

**Issue by 8bitFra** [Tkd-Alex/Twitch-Channel-Points-Miner-v2#53]
**Bug**
I cannot get the software to run on an operating system without a desktop environment. I think I have configured everything correctly, any idea what it might be?

**To Reproduce**
In order to reproduce the behavior you just need to run the software on an operating system without a desktop enviroment with the default configuration.

**Useful informations**
 - OS: Ubuntu server 20.04
 - Browser: Chrome
 - Python version: 3.x

**Screen Error**
![error github](https://user-images.githubusercontent.com/47214887/107069308-135c0680-67e2-11eb-8ef3-3810c0adb8a5.png)

**Comment by RakSrinaNa** [comment]
To me it sounds like a problem on your server and isn't related to this project in particular.
Can you run other programs that uses a browser driver in headless mode?

We're several running it headless without a desktop environment and it works fine.

**Comment by RakSrinaNa** [comment]
Also from personal experience, I've had a lot of problem with packages coming from snap. Especially regarding permissions. Maybe you could try getting the driver from a more "standard way" with `apt install chromium-chromedriver`

**Comment by Tkd-Alex** [comment]
You can try also to download the chromedriver from here: https://chromedriver.chromium.org/ and put as argument of BrowserSettings.
Or test: https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/pull/50 :hand_over_mouth:

**Comment by RakSrinaNa** [comment]
To confirm, I tried installing an Ubuntu 20.04 server, added the `chromium-chromedriver` package and it loads fine.

Though as Tkd said you can probably try the linked issue to avoid having to setup a webdriver.

**Comment by Element21** [comment]
Use firefox and geckodriver. If you install Firefox, at least on archlinux,
it only installs a few fonts but no x server or anything surprisingly. The
dev is working on a version of this program without selenium (look at the
off-selenium branch).

**Comment by 8bitFra** [comment]
> To confirm, I tried installing an Ubuntu 20.04 server, added the `chromium-chromedriver` package and it loads fine.
> 
> Though as Tkd said you can probably try the linked issue to avoid having to setup a webdriver.

I've tried with this setup before, but I don't know why I can't get it to work with Chromium. I installed the binaries written in this issue (https://github.com/Tkd-Alex/Twitch-Channel-Points-Miner-v2/issues/32) and couldn't get it to work without desktop-enviroment (with desktop enviroment all it works properly). Now making other attempts I solved it by making it work with firefox and geckodriver (and to be honest it was a pretty embarrassing mistake ahaha). However I still don't understand why it doesn't work for me with chromium as you claim it works correctly on your instance, but I can consider the problem as solved.
